tools/project_manager/update_illos.php

```diff
@@ -0,0 +1,210 @@
+    else
+    {
+        echo "<p>";
+        echo _('An error occurred.'), "\n";
+        echo $err_msg, "\n";
+        echo _('Please try again.'), "\n";
+        echo "</p>\n";
+        // Fall through to try again.
+    }
+}
+
+if ($operation == 'replace') {
+    echo "<p>", _('Select a replacement image to upload:'), "</p>\n";
+    echo "
+        <form enctype='multipart/form-data' action='update_illos.php' method='post'>
+        <input type='hidden' name='projectid' value='$projectid'>
+        <input type='hidden' name='image' value='$image'>
+        <input type='hidden' name='operation' value='$operation'>
+        <input type='file' name='replacement_image' size='50'>
+        <br>
+        <input type='submit' value='", attr_safe(_("Upload Image")), "'>
+        </form>
+    ";
+} else {
+    echo "<p>", _('Are you sure you want to delete this image?'), "</p>\n";
+    echo "
+        <form enctype='multipart/form-data' action='update_illos.php' method='post'>
+        <input type='hidden' name='projectid' value='$projectid'>
+        <input type='hidden' name='image' value='$image'>
+        <input type='hidden' name='operation' value='$operation'>
+        <input type='hidden' name='confirmed' value='yes'>
+        <br>
+        <input type='submit' value='", attr_safe(_("Do It")), "'>
+        </form>
+    ";
+}
+
+// XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+
+function handle_upload( $replacement_image_info )
+// If there's a problem, return a string containing an error message.
+// If no problem, return the empty string.
+{
+    global $projectid, $image;
+    global $projects_dir;
+
+    // Check the error code.
+    $error_code = $replacement_image_info['error'];
+    if ($error_code != UPLOAD_ERR_OK)
+    {
+        return
+            sprintf( _('Error code = %d.'), $error_code )
+            . "\n"
+            . "(" . get_upload_err_msg($error_code) . ")";
+    }
+
+    // Check that the extensions match.
+    $curr_ext = pathinfo($image, PATHINFO_EXTENSION);
+    $repl_ext = pathinfo($replacement_image_info['name'], PATHINFO_EXTENSION);
+    if ( $curr_ext != $repl_ext )
+    {
+        return sprintf(
+            _('Replacement file\'s extension (%1$s) does not match current file\'s extension (%2$s).'),
+            $repl_ext,
+            $curr_ext
+        );
+    }
+
+    // $replacement_image_info['type'] might be relevant, but is it trustworthy?
+
+    // Check $replacement_image_info['size']?
+
+    $image_path = "$projects_dir/$projectid/$image";
+    $r = move_uploaded_file($replacement_image_info['tmp_name'], $image_path);
+    if ( $r )
+    {
+        return '';
+    }
+    else
+    {
+        return _('The uploaded file cannot be moved into the project directory for some reason.');
+    }
+}
+
+function handle_delete()
+// If there's a problem, return a string containing an error message.
+// If no problem, return the empty string.
+{
+    global $projectid, $image;
```

Comment:
We should pass these two global variables into the function instead of having them be globals.